pyramid_oereb/contrib/print_proxy/mapfish_print/toc_pages.py

```diff
@@ -5,116 +5,206 @@
+        )
+        if total_size > self.d6_left_height:
+            log.debug("d6 left total_size: {}".format(total_size))
             return total_size
         else:
-            return content_min_size
+            log.debug("d6 left total_size: {}".format(self.d6_left_height))
+            return self.d6_left_height
 
     @staticmethod
-    def compute_length_of_wrapped_text(text, nb_char, font_size):
+    def compute_length_of_wrapped_text(text, nb_char, line_height):
         t = textwrap.wrap(text, nb_char)
-        return len(t) * font_size
+        return len(t) * line_height
 
     def compute_d6_right(self):
-        # variables taken from template disclaimer.jrxml
-        space_above = 4
-        space_title_content = 2
-        content_min_size = 23
         total_size = 0
-        for i in self.extract.get('Disclaimer', []):
-            total_size += space_above
-            total_size += self.compute_length_of_wrapped_text(i['Title'][0]['Text'],
-                                                              65,
-                                                              14)
-            total_size += space_title_content
-            total_size += self.compute_length_of_wrapped_text(i['Content'][0]['Text'],
-                                                              78,
-                                                              10)
-        log.debug('d6 ritght total_size : {}'.format(total_size))
-        if total_size > content_min_size:
+
+        blank_space_below_disclaimers = 6  # disclaimer_and_qrcode.jrxml
+        disclaimer_title_line_height = 8  # disclaimer_and_qrcode.jrxml
+        disclaimer_item_line_height = 8  # disclaimer_and_qrcode.jrxml
+        blank_space_above_qrcode = 13  # disclaimer_and_qrcode.jrxml
+        qr_code_size = 56  # disclaimer_and_qrcode.jrxml
+
+        # Disclaimers (multiple items)
+        for i in self.extract.get("Disclaimer", []):
+            total_size += self.compute_length_of_wrapped_text(
+                i["Title"], 65, disclaimer_title_line_height
+            )
+            total_size += self.compute_length_of_wrapped_text(
+                i["Content"], 73, disclaimer_item_line_height
+            )
+            total_size += blank_space_below_disclaimers
+
+        # QR-Code (optional)
+        if self.display_qrcode:
+            total_size += blank_space_above_qrcode + qr_code_size
+
+        if total_size > self.d6_right_height:
+            log.debug("d6 right total_size: {}".format(total_size))
             return total_size
         else:
-            return content_min_size
+            log.debug("d6 right total_size: {}".format(self.d6_right_height))
+            return self.d6_right_height
 
     def compute_d6(self):
-        x = max(self.compute_d6_left(), self.compute_d6_right()) + self.d6_stuff_y_location
-        if x > self.d6_height:
-            return x
+        total_size = max(self.compute_d6_left(), self.compute_d6_right())
```

Comment:
D6 also has a top margin of 26px. Did you take that into account?

Reply:
No, I didn't. I made a bugfix. Thanks for the hint.